engine-tests/src/test/java/org/terasology/engine/rendering/assets/mesh/VertexGLAttributeTest.java

```diff
@@ -0,0 +1,259 @@
+import org.terasology.engine.rendering.assets.mesh.resource.VertexResource;
+import org.terasology.engine.rendering.assets.mesh.resource.VertexResourceBuilder;
+import org.terasology.nui.Color;
+import org.terasology.nui.Colorc;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class VertexGLAttributeTest {
+
+    @Test
+    public void testFloatBinding() {
+        VertexResourceBuilder builder = new VertexResourceBuilder();
+        VertexFloatAttributeBinding a1 = builder.add(0, GLAttributes.FLOAT_1_VERTEX_ATTRIBUTE);
+        VertexResource resource = builder.build();
+
+        a1.put(10.0f);
+        a1.put(15.0f);
+        a1.put(15.5f);
+        a1.put(2.0f);
+        a1.put(-100.0f);
+
+        assertEquals(5, a1.getPosition());
+        resource.writeBuffer(buffer -> {
+            assertEquals(5 * Float.BYTES, buffer.limit());
+
+            assertEquals(10.0f, buffer.getFloat(Float.BYTES * 0), 0.0001f);
+            assertEquals(15.0f, buffer.getFloat(Float.BYTES * 1), 0.0001f);
+            assertEquals(15.5f, buffer.getFloat(Float.BYTES * 2), 0.0001f);
+            assertEquals(2.0f, buffer.getFloat(Float.BYTES * 3), 0.0001f);
+            assertEquals(-100.0f, buffer.getFloat(Float.BYTES * 4), 0.0001f);
+        });
+    }
+
+    @Test
+    public void testIntBinding() {
+        VertexResourceBuilder builder = new VertexResourceBuilder();
+        VertexIntegerAttributeBinding a1 = builder.add(0, GLAttributes.INT_1_VERTEX_ATTRIBUTE);
+        VertexResource resource = builder.build();
+
+        a1.put(10);
+        a1.put(20);
+        a1.put(150);
+        a1.put(100);
+        a1.put(-100);
+
+        assertEquals(5, a1.getPosition());
+        resource.writeBuffer(buffer -> {
+            assertEquals(5 * Integer.BYTES, buffer.limit());
+
+            assertEquals(10, buffer.getInt(Integer.BYTES * 0));
+            assertEquals(20, buffer.getInt(Integer.BYTES * 1));
+            assertEquals(150, buffer.getInt(Integer.BYTES * 2));
+            assertEquals(100, buffer.getInt(Integer.BYTES * 3));
+            assertEquals(-100, buffer.getInt(Integer.BYTES * 4));
+        });
+    }
+
+    @Test
+    public void testByteBinding() {
+        VertexResourceBuilder builder = new VertexResourceBuilder();
+        VertexIntegerAttributeBinding a1 = builder.add(0, GLAttributes.BYTE_1_VERTEX_ATTRIBUTE);
+        VertexResource resource = builder.build();
+
+        a1.put(10);
+        a1.put(150);
+        a1.put(300);
+        a1.put(100);
+
+        assertEquals(4, a1.getPosition());
+        resource.writeBuffer(buffer -> {
+            assertEquals(4 * Byte.BYTES, buffer.limit());
+
+            assertEquals(10, Byte.toUnsignedInt(buffer.get(Byte.BYTES * 0)));
+            assertEquals(150, Byte.toUnsignedInt(buffer.get(Byte.BYTES * 1)));
+            assertEquals(255, Byte.toUnsignedInt(buffer.get(Byte.BYTES * 2)));
```

Comment:
Ah, this `255` case is the one that goes with that `clamp()` you added. I suggest pointing that out with a comment, it's pretty easy to miss when scanning through these lines and numbers that are mostly all the same.

Should this test a negative value too?